src/dayNightSystem/dayNightClock.mjs

```diff
@@ -6,69 +6,101 @@
 // note: don't run more than one day/night clock at a time.  doing so will cause multiple
 //       filters to be applied to the screen, which won't look too nice. :o)
 
-import { Prim, Thread } from 'sphere-runtime';
+import { Thread } from 'sphere-runtime';
 
 import InGameTime from './inGameTime';
 
-const DayMask = Color.Transparent,
-      TwilightMask = new Color(0.5, 0.125, 0.0625, 0.625),
-      NightMask = new Color(0.0, 0.0, 0.125, 0.5625);
+const DayMask = [0.0, 0.0, 0.0, 0.0],
+      TwilightMask = [0.5, 0.125, 0.0625, 0.625],
+      NightMask = [0.0, 0.0, 0.125, 0.5625];
+
+const Day = 0,
+      Night = 1,
+      Fading = 2;
 
 export default
 class DayNightClock extends Thread
 {
 	constructor()
 	{
 		super({ priority: 1 });
+		this.inGameTime = new InGameTime(0, 0, 0);
 
 		console.log("initializing day/night clock", `time: ${this.now()}`);
-		this.currentMask = Color.Transparent;
+		let white = Color.White;
+		let width = Surface.Screen.width;
+		let height = Surface.Screen.height;
+		let rectangle = new Shape(ShapeType.TriStrip,
+			new VertexList([
+				{ x: 0, y: 0, color: white },
+				{ x: width, y: 0, color: white },
+				{ x: 0, y: height, color: white },
+				{ x: width, y: height, color: white },
+			]));
+		this.model = new Model ([rectangle]);
+		this.shader = new Shader({
+			fragmentFile: '#/shaders/image.frag.glsl',
+			vertexFile  : '#/shaders/image.vert.glsl',
+		});
+		this.model.shader = this.shader;
```

Comment:
Maybe we should just create the shader before the model and do `new Model([ rectangle ], shader);` so we don't need to set it afterwards.

Reply:
done